hal/armv7m/imxrt/10xx/hal.c

```diff
@@ -66,6 +62,12 @@ void hal_done(void)
 }
 
 
+void hal_syspageSet(addr_t addr)
+{
+	hal_common.hs = (hal_syspage_t *)addr;
```

Comment:
Let's pass hal_syspage_t pointer from the start
```suggestion
void hal_syspageSet(hal_syspage_t  *addr)
{
	hal_common.hs = addr;
```